src/activity_handlers.cpp

```diff
@@ -2717,6 +2718,21 @@ void activity_handlers::butcher_do_turn( player_activity *, player *p )
     p->mod_stat( "stamina", -20.0f * p->stamina / p->get_stamina_max() );
 }
 
+void activity_handlers::read_do_turn( player_activity *act, player *p )
+{
+    if( !act->str_values.empty() && act->str_values[0] == "martial_art" && one_in( 3 ) ) {
+        if( act->values.size() == 0 ) {
+            act->values.push_back( p->stamina );
+        }
+        p->stamina = act->values[0] - 1;
+        act->values[0] = p->stamina;
+    }
+    if( p->stamina < p->get_stamina_max() * 0.1 ) {
```

Comment:
```suggestion
    if( p->stamina < p->get_stamina_max() / 10 ) {
```
No need to risk floating point oddities here.